STM32F405RGTx/Core/Src/main.cpp

```diff
@@ -744,7 +783,9 @@ void HAL_TIM_PeriodElapsedCallback(TIM_HandleTypeDef *htim)
     HAL_IncTick();
   }
   /* USER CODE BEGIN Callback 1 */
-
+  else if (htim->Instance == TIM7) {
+    // TODO: LED blinking (call function from state_machine thread)
```

Comment:
You can call the state machine function here now since you added it. Rename it as mentioned in the other comment first tho but it would be something like `StateLEDCallback()`